src/KeyVault/KeyVault/Commands/NewAzureKeyVault.cs

```diff
@@ -70,54 +75,87 @@ public class NewAzureKeyVault : KeyVaultManagementCmdletBase
         [ValidateNotNullOrEmpty]
         public int SoftDeleteRetentionInDays { get; set; }
 
         [Parameter(Mandatory = false,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "Specifies the SKU of the key vault instance. For information about which features are available for each SKU, see the Azure Key Vault Pricing website (http://go.microsoft.com/fwlink/?linkid=512521).")]
-        public SkuName Sku { get; set; }
+        public string Sku { get; set; }
 
         [Parameter(Mandatory = false,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "A hash table which represents resource tags.")]
         [Alias(Constants.TagsAlias)]
         public Hashtable Tag { get; set; }
 
-        [Parameter(Mandatory = false, HelpMessage = "Specifies the network rule set of the vault. It governs the accessibility of the key vault from specific network locations. Created by `New-AzKeyVaultNetworkRuleSetObject`.")]
+        [Parameter(Mandatory = false,
+            // Hide out until available
+            ParameterSetName = KeyVaultParameterSet,
+            HelpMessage = "Specifies the network rule set of the vault. It governs the accessibility of the key vault from specific network locations. Created by `New-AzKeyVaultNetworkRuleSetObject`.")]
         public PSKeyVaultNetworkRuleSet NetworkRuleSet { get; set; }
 
         #endregion
 
-        public override void ExecuteCmdlet()
+        #region Keyvault-specified Parameter Definitions
+
+        [Parameter(Mandatory = false,
+            ParameterSetName = KeyVaultParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "If specified, enables secrets to be retrieved from this key vault by the Microsoft.Compute resource provider when referenced in resource creation.")]
+        public SwitchParameter EnabledForDeployment { get; set; }
+
+        [Parameter(Mandatory = false,
+            ParameterSetName = KeyVaultParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "If specified, enables secrets to be retrieved from this key vault by Azure Resource Manager when referenced in templates.")]
+        public SwitchParameter EnabledForTemplateDeployment { get; set; }
+
+        [Parameter(Mandatory = false,
+            ParameterSetName = KeyVaultParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "If specified, enables secrets to be retrieved from this key vault by Azure Disk Encryption.")]
+        public SwitchParameter EnabledForDiskEncryption { get; set; }
+
+        #endregion
+
+        #region MHSM-specified Parameter Definitions
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = ManagedHsmParameterSet,
+            HelpMessage = "Array of initial administrators object ids for this managed hsm pool.")]
+        public List<string> Administrator { get; set; }
```

Comment:
We should use array instead of generic collection types. See: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/parameter-best-practices.md#array-vs-enumerable-types

I didn't notice this during the design review, sorry

```suggestion
        public string[] Administrator { get; set; }
```